arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/control/JoystickSeekBar.kt

```diff
@@ -0,0 +1,231 @@
+
+    init {
+        // Set the initial progress to half of the max value
+        progress = (max * 0.5).roundToInt()
+        setOnSeekBarChangeListener(this)
+    }
+
+    /**
+     * Constructor used when instantiating this View directly to attach it to another view
+     * programmatically.
+     *
+     * @since 100.6.0
+     */
+    constructor(context: Context) : super(context)
+
+    /**
+     * Constructor used when defining this view in an XML layout.
+     *
+     * @since 100.6.0
+     */
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        context.theme.obtainStyledAttributes(
+            attrs,
+            R.styleable.JoystickSeekBar,
+            0, 0
+        ).apply {
+            try {
+                _min = getFloat(
+                    R.styleable.JoystickSeekBar_jsb_min,
+                    DEFAULT_MIN
+                )
+                _max = getFloat(
+                    R.styleable.JoystickSeekBar_jsb_max,
+                    DEFAULT_MAX
+                )
+            } finally {
+                recycle()
+            }
+        }
+        if (_min >= _max) {
+            throw RuntimeException("Attribute jsb_min must be less than attribute jsb_max")
+        }
+    }
+
+    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
+        deltaProgress = offsetProgress.pow(2) / 10.0f * (if (offsetProgress < 0) -1.0f else 1.0f)
+    }
+
+    override fun onStartTrackingTouch(seekBar: SeekBar?) {
+        deltaTimerTask = object : TimerTask() {
+            override fun run() {
+                post {
+                    deltaProgressUpdatedListeners.forEach {
+                        it.onDeltaProgressUpdated(deltaProgress)
+                    }
+                }
+            }
+        }
+        deltaTimer.schedule(deltaTimerTask, 0, DEFAULT_DELTA_INTERVAL_MILLIS)
+    }
+
+    override fun onStopTrackingTouch(seekBar: SeekBar?) {
+        deltaTimerTask.cancel()
+        deltaProgress = 0.0f
+        progress = (max * 0.5).roundToInt()
+    }
+
+    /**
+     * Add a [listener] to be notified when a [deltaProgress] update is made. Returns true if the
+     * listener was added, false otherwise.
+     *
+     * @since 100.6.0
+     */
+    fun addDeltaProgressUpdatedListener(listener: DeltaProgressUpdatedListener): Boolean {
+        return this.deltaProgressUpdatedListeners.add(listener)
+    }
+
+    /**
+     * Removes a previously added [listener]. Returns true if the listener was removed, false
+     * otherwise.
+     *
+     * @since 100.6.0
+     */
+    fun removeDeltaProgressUpdatedListener(listener: DeltaProgressUpdatedListener): Boolean {
+        return this.removeDeltaProgressUpdatedListener(listener)
+    }
+
+    /**
+     * Listener interface to listen for delta progress updates.
+     *
+     * @since 100.6.0
+     */
+    interface DeltaProgressUpdatedListener {
+        fun onDeltaProgressUpdated(deltaProgress: Float)
```

Comment:
Might be good to add some doc explaining what `deltaProgress` means.

Reply:
I described it at the top of the JoystickSeekBar, would that suffice?

```
The distance from the center point of the slider determines the [deltaProgress] value applied during each interval. For distances close to the center, the delta is small and used for fine-grained adjustments; distances farther from the center are for larger-scale adjustments. When released, the slider "thumb" will return to the center of the slider.
```

Reply:
I think it would be good to add that doc right here, as this is where a client of that interface would look for that information.